streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -424,9 +460,41 @@ public Subscription subscription(Set<String> topics) {
         // change log topics should be compacted
         prepareTopic(stateChangelogTopicToTaskIds, true /* compactTopic */, true);
 
+        Map<String, Assignment> assignment = new HashMap<>();
+        for (AssignmentDetail assignmentDetail : assignmentDetails) {
+            assignment.put(assignmentDetail.consumer, assignmentDetail.createAssignment());
+        }
         return assignment;
     }
 
+    class AssignmentDetail {
```

Comment:
I'd rather not as `AssignmentInfo` is really just the over the wire format whereas this class is collecting the data needed to create an `Assignment` with `AssignmentInfo`. Also, `AssignmentInfo` doesn't need the list of `activePartitions` or the `consumer` so it would be a bit strange to add them to that class and then non encode/decode them as they don't need to be. I've renamed the class to `AssignmentSupplier` which hopefully makes it a bit more obvious about what it is doing.